aquadoggo/src/materializer/tasks/blob.rs

```diff
@@ -75,38 +73,36 @@ pub async fn blob_task(context: Context, input: TaskInput) -> TaskResult<TaskInp
             // We don't raise a critical error here, as it is possible that this method returns an
             // error.
             .map_err(|err| TaskError::Failure(err.to_string()))?
-            .unwrap();
+            .expect("Blob data exists at this point");
 
         // Compose, and when needed create, the path for the blob file.
         let base_path = match &context.config.blob_dir {
             Some(base_path) => base_path,
             None => return Err(TaskError::Critical("No base path configured".to_string())),
         };
-
-        let blob_dir = base_path.join(blob_document.id().as_str());
-
-        fs::create_dir_all(&blob_dir).map_err(|err| TaskError::Critical(err.to_string()))?;
-        let blob_view_path = blob_dir.join(blob_document.view_id().to_string());
+        // @TODO: Why is this required here? Shouldn't the folder already exist at this point? (if
+        // I remove this writing a file fails ..)
+        fs::create_dir_all(base_path).unwrap();
```

Comment:
This `create_dir_all` is a little bit of a mystery for me. If I remove it, creating a file fails, even though the folder should have been already created when the whole node got initialised?

Reply:
I would like to explore this a bit more when we have the configuration interface (where `base_path` might change into something outside of `/tmp`)

Reply:
Change this now to an async tokio version of this method .. that seems to do the trick?